src/features/vote/Wallet.tsx

```diff
@@ -3,18 +3,61 @@ import { FC } from "react";
 import tw, { styled } from "twin.macro"; // eslint-disable-line
 import Modal from "common/components/modal";
 import useModal from "common/hooks/useModal";
-
+import Button from "common/components/button";
+import useConnection from "common/hooks/useConnection";
 import { Settings } from "assets/icons";
 
-const Wallet: FC = () => {
+interface Props {
+  // connect: Connect;
+  // disconnect: Disconnect;
+}
+
+const Wallet: FC<Props> = () => {
   const { isOpen, open, close, modalRef } = useModal();
+  const {
+    initOnboard,
+    setInitOnboard,
+    isConnected,
+    onboard,
+    disconnect,
+  } = useConnection();
+
   return (
     <StyledWallet>
       <div tw="flex items-stretch items-center">
         <div tw="py-8 pl-5 flex-grow">
           <p className="wallet-title">Voting Wallet</p>
-          <Connected>Connected with MetaMask</Connected>
+          {isConnected ? (
+            <Connected>
+              Connected with {onboard?.getState().wallet.name}
+            </Connected>
+          ) : (
+            <Disconnected>Not Connected</Disconnected>
+          )}
+
+          {!isConnected ? (
+            <Button
+              className="connect-btn"
+              onClick={() => {
+                if (!initOnboard) setInitOnboard(true);
+              }}
+              variant="primary"
+            >
+              Connect Wallet
+            </Button>
+          ) : (
+            <Button
+              className="connect-btn"
```

Comment:
I think that styling stuff like this will bite us sooner or later, as it leaves it on us to come up with unique class names / deal with specificity, for common components like Buttons or Headers etc... 
Not blocking as it will get resolved once we have a component library for more common components, but something to keep in mind imho.